Sources/NonEmpty/NonEmpty.swift

```diff
@@ -10,8 +10,9 @@ public struct NonEmpty<Collection: Swift.Collection>: Swift.Collection {
     self.rawValue = rawValue
   }
 
-  public subscript<Subject>(dynamicMember keyPath: KeyPath<Collection, Subject>) -> Subject {
-    self.rawValue[keyPath: keyPath]
+  public subscript<Subject>(dynamicMember keyPath: WritableKeyPath<Collection, Subject>) -> Subject {
+    _read { yield self.rawValue[keyPath: keyPath] }
+    _modify { yield &self.rawValue[keyPath: keyPath] }
```

Comment:
why `_read` and `_modify` rather than `get` and `set`?

Reply:
We prefer `_read` and `_modify` here for performance reasons, as it allows direct read/write access to the underlying value.